Hypriot-OS-Course-Preparation-Instructions.md

```diff
@@ -0,0 +1,75 @@
+## Raspberry-Pi based Akka Cluster workshop preparation instructions
+
+In case you're attending a `Raspberry-Pi based Akka Cluster workshop`, you will need to prepare you're laptop to integrate with a pre-installed 5-node cluster you'll be working on.
+
+The following sections give instructions on how to achieve this.
+
+### Adding the cluster nodes to /etc/hosts
+
+On your Mac, add the following entries to your _/etc/hosts_ file:
+
+```
+# Cluster #0
+192.168.200.10 node-0
+192.168.200.11 node-1
+192.168.200.12 node-2
+192.168.200.13 node-3
+192.168.200.14 node-4
+```
+
+
+### Configure password-less login
+
+Next, we set-up password-less login. We assume that we want to log in to account _akkapi_ on a pi from an account (_userxxx_) on your laptop. In order to set-up password-less login, you need a so-called public/private key pair. If you're unfamiliar with this concept, or don't have such a key pair, first read the paragraph titled _Generating a public/private key pair_ in the Addendum.
+
+Now, proceed by create a `.ssh` folder on the _akkapi_'s home folder on the Pi and copy the public key (`id_rsa.pub`) to a file named `authorized_keys` in the `.ssh` folder. You will have to supply the password for the _akkapi_ account for the two commands launched. On your laptop:
+
+```
+[Pi-Akka-Cluster git:(master) ✗ for node in 0 1 2 3 4;do ssh akkapi@node-${node} mkdir .ssh; done
+akkapi@node-0's password:
+```
+
+Copy the contents of your *public key* to the `authorized_keys` file:
+
+```
+Pi-Akka-Cluster git:(master) ✗ for node in 0 1 2 3 4;do cat ~/.ssh/id_rsa.pub | ssh akkapi@node-${node} 'cat >> .ssh/authorized_keys'; done
+akkapi@node-0's password:
+```
+
+With this, you should now be able to log into the _akkapi_ on all nodes from the _userxxx_ account on your laptop without having to enter a password.
+
```

Comment:
I think I've seen this work for people without copying their public key as well - probably their gpg-agent remembered the password after entering it the first time?

:+1: to document this approach - perhaps prefix with "if your ssh does not remember your password, you can optionally do this' to save people who are already set up some work?

Reply:
That's very interesting! I never used a gpg-agent. It probably also would solve an annoying issue I encounter regularly when working with multiple clusters: when switching between clusters, I need to clean out entries in `~/.ssh/known_hosts`.

Reply:
@raboof Indeed, that's the plan; actually there's one open PR from @kikiya that aims to do just that. It needs to be adapted a bit as it assumes working with a studentified version of the master repo. This is the long term plan, but right now, we work with the master repo.

Reply:
@raboof re-phrased instructions based on your comment.
 
> 👍 to document this approach - perhaps prefix with "if your ssh does not remember your password, you can optionally do this' to save people who are already set up some work?